src/prefect/tasks/redis/redis_tasks.py

```diff
@@ -0,0 +1,214 @@
+        - redis_key (optional): Redis key to be set, can be provided at initialization or runtime
+        - redis_val (optional): Redis val to be set, can be provided at initialization or runtime
+        - ex (int, optional): if provided, sets an expire flag, in seconds, on 'redis_key' set
+        - px (int, optional): if provided, sets an expire flag, in milliseconds, on 'redis_key' set
+        - nx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+            if it does not exist, defaults to False
+        - xx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+            if it already exists, defaults to False
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        connection=None,
+        host: str = "localhost",
+        port: int = 6379,
+        db: int = 0,
+        password: str = None,
+        redis_key=None,
+        redis_val=None,
+        ex: int = None,
+        px: int = None,
+        nx: bool = False,
+        xx: bool = False,
+        **kwargs
+    ):
+        ## connect to Redis
+        if connection:
+            self.connection = connection
+        else:
+            self.connection = redis.Redis(
+                host=host, port=port, db=db, password=password
+            )
+
+        self.redis_key = redis_key
+        self.redis_val = redis_val
+        self.ex = ex
+        self.px = px
+        self.nx = nx
+        self.xx = xx
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("redis_key", "redis_val", "ex", "px", "nx", "xx")
+    def run(
+        self,
+        redis_key=None,
+        redis_val=None,
+        ex: int = None,
+        px: int = None,
+        nx: bool = False,
+        xx: bool = False,
+    ):
+        """
+        Task run method. Sets Redis key-value pair.
+
+        Args:
+            - redis_key (optional): Redis key to be set, can be provided at initialization or runtime
+            - redis_val (optional): Redis val to be set, can be provided at initialization or runtime
+            - ex (int, optional): if provided, sets an expire flag, in seconds, on 'redis_key' set
+            - px (int, optional): if provided, sets an expire flag, in milliseconds, on 'redis_key' set
+            - nx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+                if it does not exist, defaults to False
+            - xx (int, optional): if set to True, set the value at 'redis_key' to 'redis_val' only
+                if it already exists, defaults to False
+
+        Returns:
+            - bool: status of set operation
+
+        Raises:
+            - ValueError: if redis_key or redis_val is not provided
+        """
+        if None in (redis_key, redis_val):
+            raise ValueError("redis_key and redis_val must be provided")
```

Comment:
Nice.

Reply:
Thanks! I was unreasonably pleased with this line haha